crypto/src/vrf/schnorrkel/mod.rs

```diff
@@ -0,0 +1,158 @@
+use crate::random::{CryptoRng, Rng};
+use parity_scale_codec::{Decode, Encode};
+
+const EXPECTED_PUBKEY_LEN: usize = 32;
+
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[must_use]
+pub struct SchnorrkelPublicKey {
+    key: schnorrkel::PublicKey,
+}
+
+impl Encode for SchnorrkelPublicKey {
+    fn encode(&self) -> Vec<u8> {
+        self.key.to_bytes().to_vec()
+    }
+    fn encoded_size(&self) -> usize {
+        debug_assert_eq!(self.key.to_bytes().len(), EXPECTED_PUBKEY_LEN);
+        EXPECTED_PUBKEY_LEN
+    }
+}
+
+impl Decode for SchnorrkelPublicKey {
+    fn encoded_fixed_size() -> Option<usize> {
+        Some(EXPECTED_PUBKEY_LEN)
+    }
+
+    fn decode<I: parity_scale_codec::Input>(
+        input: &mut I,
+    ) -> Result<Self, parity_scale_codec::Error> {
+        const ERR_MSG: &str = "Failed to read schnorrkel public key";
+        let mut v = [0; EXPECTED_PUBKEY_LEN];
+        input.read(v.as_mut_slice())?;
+        let key = schnorrkel::PublicKey::from_bytes(&v)
+            .map_err(|_| parity_scale_codec::Error::from(ERR_MSG))?;
+        Ok(Self { key })
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+#[must_use]
+pub struct SchnorrkelPrivateKey {
+    key: schnorrkel::SecretKey,
+}
+
+impl SchnorrkelPrivateKey {
+    #[allow(dead_code)]
+    pub fn new<R: Rng + CryptoRng>(rng: &mut R) -> (SchnorrkelPrivateKey, SchnorrkelPublicKey) {
+        let sk = schnorrkel::SecretKey::generate_with(rng);
+        let pk = sk.to_public();
+        let sk = Self { key: sk };
+        let pk = SchnorrkelPublicKey { key: pk };
+        (sk, pk)
+    }
```

Comment:
It may be useful to expose `to_public` as a stand-alone function as well. Or are there reasons to not do it?

Reply:
This is still incomplete